src/vit-servicing-station/vit-servicing-station-tests/src/tests/raw_snapshot.rs

```diff
@@ -0,0 +1,361 @@
+            )
+            .unwrap();
+        assert_eq!(
+            vec![voting_power],
+            voter_info.voter_info,
+            "expected non-empty data for entry idx: {}",
+            idx
+        );
+        assert_eq!(
+            second_raw_snapshot.content.update_timestamp, voter_info.last_updated,
+            "wrong timestamp for entry idx: {}",
+            idx
+        );
+    }
+
+    let third_snapshot = SnapshotBuilder::default()
+        .with_timestamp(second_raw_snapshot.content.update_timestamp + 1)
+        .with_tag(second_raw_snapshot.tag.clone())
+        .build();
+
+    rest_client.put_snapshot_info(&third_snapshot).unwrap();
+
+    for (idx, entry) in second_snapshot.iter().enumerate() {
+        let voter_info = rest_client
+            .voter_info(&second_raw_snapshot.tag, &entry.hir.voting_key.to_hex())
+            .unwrap();
+        assert!(
+            voter_info.voter_info.is_empty(),
+            "expected empty data for entry idx: {}",
+            idx
+        );
+        assert_eq!(
+            third_snapshot.content.update_timestamp, voter_info.last_updated,
+            "wrong timestamp for entry idx: {}",
+            idx
+        );
+    }
+
+    for (idx, entry) in third_snapshot.content.snapshot.iter().enumerate() {
+        let voting_power = VotingPower::from(entry.clone());
+        let voter_info = rest_client
+            .voter_info(&third_snapshot.tag, &entry.hir.voting_key.to_hex())
+            .unwrap();
+        assert_eq!(
+            vec![voting_power],
+            voter_info.voter_info,
+            "expected non-empty data for entry idx: {}",
+            idx
+        );
+        assert_eq!(
+            third_snapshot.content.update_timestamp, voter_info.last_updated,
+            "wrong timestamp for entry idx: {}",
+            idx
+        );
+    }
+}
+
+#[test]
+pub fn import_raw_snapshots_with_different_tags() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let rest_client = server.rest_client_with_token(&data.token_hash());
+
+    let first_raw_snapshot = RawSnapshot::default();
+
+    rest_client.put_raw_snapshot(&first_raw_snapshot).unwrap();
+
+    let second_raw_snapshot = RawSnapshotUpdater::from(first_raw_snapshot.clone())
+        .with_tag("fund9")
+        .build();
+
+    rest_client.put_raw_snapshot(&second_raw_snapshot).unwrap();
+
+    let assigner = RepsVotersAssigner::new(
+        first_raw_snapshot.content.direct_voters_group.unwrap(),
+        first_raw_snapshot.content.representatives_group.unwrap(),
+    );
+
+    let first_snapshot = Snapshot::from_raw_snapshot(
+        first_raw_snapshot.content.snapshot,
+        first_raw_snapshot.content.min_stake_threshold,
+        first_raw_snapshot.content.voting_power_cap,
+        &assigner,
+    )
+    .unwrap()
+    .to_full_snapshot_info();
+
+    for (idx, entry) in first_snapshot.iter().enumerate() {
```

Comment:
There is some duplication across verifications. I would personally introduce some method so it can be more readable what is going on. like :

`
```
fn assert_against_snapshot(snapshot_entry: (..), raw_snapshot: RawSnapshot) {
    let voting_power = VotingPower::from(snapshot_entry.clone());
        let voter_info = rest_client
            .voter_info(&first_raw_snapshot.tag, &entry.hir.voting_key.to_hex())
            .unwrap();
        assert_eq!(
            vec![voting_power.clone()],
            voter_info.voter_info,
            "wrong data for entry idx: {}",
            idx
        );
        assert_eq!(
            raw_snapshot.content.update_timestamp, voter_info.last_updated,
            "wrong timestamp for entry idx: {}",
            idx
        );
}
```

Reply:
done